react/week2/to_do_list/src/index.js

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const taskList = [
+  {
+    id: 0,
+    description: "Get out of bed",
+    color: "brown",
+    deadline: "2020-06-08",
+    done: false,
+    visibility: true
+  },
+  {
+    id: 1,
+    description: "Brush teeth",
+    color: "red",
+    deadline: "2020-06-08",
+    done: false,
+    visibility: true
+  },
+  {
+    id: 2,
+    description: "Eat breakfast",
+    color: "orange",
+    deadline: "2020-06-08",
+    done: true,
+    visibility: true
+  }
+];
+
+class Header extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1> Welcome to My ToDo List</h1>
+        <br ></br>
+      </div>
+    )
+  }
+}
+
+class Counter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ellapsedTime: this.props.initial,
+    };
+  }
+
+  componentDidMount() {
+    setInterval(this.increment, 1000);
+  }
+
+  increment = () => {
+    this.setState({ ellapsedTime: this.state.ellapsedTime + 1 })
+  }
+
+  StartTimer = () => {
+    setInterval(this.increment, 1000);
+  }
+
+  render() {
+
+    return (
+      <div>
+        <p> you have spent {this.state.ellapsedTime} seconds in this page</p>
+        <br></br>
+      </div>
+    )
+  }
+}
+
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      toDoListData: this.props.toDoListData
```

Comment:
You should initialize state with your todolist data here instead of passing it in as props. It's ok to set default values for state!

Reply:
Do you mean that instead of all of these 4 lines just having
  state = { toDoListData: this.props.taskList }
Is that what you mean?